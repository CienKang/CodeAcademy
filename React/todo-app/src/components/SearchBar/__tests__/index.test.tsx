import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { SearchBar } from '../index';

describe('SearchBar', () => {

    it('should render an input and a button when rendered for first time', () => {

        const setTasks = jest.fn();
        const { getByPlaceholderText, getByText } = render(<SearchBar tasks={[]} setTasks={setTasks} />);
        expect(getByPlaceholderText('Enter the task name')).toBeInTheDocument();
        expect(getByText('Add')).toBeInTheDocument();
    });

    it('should add a task to the list when clicked', () => {

        const setTasks = jest.fn();
        const { getByPlaceholderText, getByText } = render(<SearchBar tasks={[]} setTasks={setTasks} />);
        const input = getByPlaceholderText('Enter the task name');
        const button = getByText('Add');
        fireEvent.change(input, { target: { value: 'test' } });
        fireEvent.click(button);
        expect(setTasks).toHaveBeenCalledWith(['test']);
    });

    it('should not add a task to the list when clicked with empty input', () => {

        const setTasks = jest.fn();
        const { getByPlaceholderText, getByText } = render(<SearchBar tasks={[]} setTasks={setTasks} />);
        const input = getByPlaceholderText('Enter the task name');
        const button = getByText('Add');
        fireEvent.change(input, { target: { value: '' } });
        fireEvent.click(button);
        expect(setTasks).not.toHaveBeenCalled();
    });

    it('should remains same when rendered for the first time', () => {

        const setTasks = jest.fn();
        const { asFragment } = render(<SearchBar tasks={[]} setTasks={setTasks} />);
        expect(asFragment()).toMatchSnapshot();
    });

});