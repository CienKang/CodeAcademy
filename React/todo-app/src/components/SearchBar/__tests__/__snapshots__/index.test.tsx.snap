// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should remains same when rendered for the first time 1`] = `
<DocumentFragment>
  <div
    class="search-bar"
  >
    <input
      placeholder="Enter the task name"
    />
    <button>
      Add
    </button>
  </div>
</DocumentFragment>
`;
